LibVLCSharp/Shared/Internal.cs

```diff
@@ -22,6 +23,11 @@ public abstract class Internal : IDisposable
         /// </summary>
         protected bool IsDisposed;
 
+        /// <summary>
+        /// The GCHandle to be passed to callbacks as userData
+        /// </summary>
+        protected GCHandle ThisHandle;
```

Comment:
Since this is only used with the `MediaPlayer` and `LibVLC` objects, maybe it could be moved down to those types? The other libvlc objects don't use it, so thats an unneeded allocation.